portality/api/v1/discovery.py

```diff
@@ -65,15 +65,14 @@ def allowed(query, wildcards=False, fuzzy=False):
 
     return True
 
-
 class DiscoveryApi(Api):
 
     @classmethod
-    def _make_query(cls, q, page, page_size, subs):
+    def _sanitise(cls, q, page, page_size, sort, search_subs, sort_subs):
         if not allowed(q):
```

Comment:
I don't think that's necessary here - "q" in these functions is the query string, not a query object, so it doesn't suffer from the same by-reference issues.  Additionally, on the subsequent lines, there is a call to 

    q = query_substitute(q, search_subs)

and this returns a totally new value for the query string, too, rather than modifying the existing one in place.

Further on still, the query itself is assembled as an object:

    query = SearchQuery(q, fro, page_size, sortby, sortdir)

and this only yields an actual query object as a dict when query.query() is called.  Even if the output of that function does get modified later on by the DAO, this won't affect the upstream object, which can generate the original query at any time by calling query.query() again (not that we need to).

In summary, yes, we should be cautious of modifying inputs in the way that the dao does (and it shouldn't do that, really), but this code insulates itself from those kinds of issues, and is well-behaved within its own scope.